app/pages/[teamSlug]/team/[userId].page.test.tsx

```diff
@@ -0,0 +1,116 @@
+import { render, screen, waitFor } from '@testing-library/react'
+import '../../../frontend/mocks/mockServer'
+import { Client, Provider } from 'urql'
+import UserDetailsPage from './[userId].page'
+import { mockMeQuery, mockUserQuery, Role } from '../../../frontend/generated/graphql'
+import { mockServer } from '../../../frontend/mocks/mockServer'
+import userEvent from '@testing-library/user-event'
+import { assignedProjects } from '../../../frontend/mocks/projectHandlers'
+
+const client = new Client({ url: '/api/team1/graphql' })
+const wrapper: React.FC = ({ children }) => <Provider value={client}>{children}</Provider>
+let userRole: Role = Role.Member
+
+jest.mock('next-auth/react', () => ({
+  useSession: () => ({ status: 'authenticated' }),
+}))
+
+jest.mock('next/router', () => ({
+  useRouter() {
+    return {
+      query: { teamSlug: 'wint', userId: '23182391283' },
+      isReady: true,
+      push: jest.fn(),
+    }
+  },
+}))
+
+beforeEach(() => {
+  mockServer.use(
+    mockMeQuery((request, response, context) => {
+      const result = response(
+        context.data({
+          __typename: 'Query',
+          user: {
+            __typename: 'User',
+            id: '',
+            image: undefined,
+            name: 'User',
+            role: Role.Admin,
+            projects: [],
+          },
+        }),
+      )
+      return result
+    }),
+  )
+
+  mockServer.use(
+    mockUserQuery((request, response, context) => {
+      const result = response(
+        context.data({
+          __typename: 'Query',
+          user: {
+            __typename: 'User',
+            id: '23182391283',
+            name: 'Test Member',
+            image: undefined,
+            role: userRole,
+            projects: assignedProjects,
+          },
+        }),
+      )
+      return result
+    }),
+  )
+})
+
+describe('UserIdPage (Admin)', () => {
+  it('should promote to admin', async () => {
+    render(<UserDetailsPage />, { wrapper })
+
+    const promoteButton = await screen.findByRole('button', {
+      name: /promote to admin/i,
+    })
+
+    await userEvent.click(promoteButton)
+    const loading = await screen.findByText('Loading...')
+    expect(loading).toBeVisible()
+    await waitFor(() => expect(loading).not.toBeInTheDocument())
+
+    userRole = Role.Admin
+  })
+
+  it('should demote to member', async () => {
+    render(<UserDetailsPage />, { wrapper })
+
+    const demoteButton = await screen.findByRole('button', {
+      name: /demote to member/i,
+    })
+
+    await userEvent.click(demoteButton)
+    const loading = await screen.findByText('Loading...')
+    expect(loading).toBeVisible()
+    await waitFor(() => expect(loading).not.toBeInTheDocument())
+  })
```

Comment:
Little hint. This test builds on the previous test.

There is the "F.I.R.S.T" principle in testing.
This test breaks the Independent/Isolated rule.

see https://medium.com/@tasdikrahman/f-i-r-s-t-principles-of-testing-1a497acda8d6